plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -209,6 +214,8 @@ public IcebergPageSourceProvider(
         this.pageIndexerFactory = requireNonNull(pageIndexerFactory, "pageIndexerFactory is null");
         requireNonNull(icebergConfig, "icebergConfig is null");
         this.maxOpenPartitions = icebergConfig.getMaxPartitionsPerWriter();
+        // this class is loaded by PluginClassLoader and we need its reference to be stored
+        this.classLoader = getClass().getClassLoader();
```

Comment:
why do you need a reference stored. Why not go to `getClass().getClassLoader()` when needed?